File: Vibe.Domain/SupportRequests/SupportRequest.cs
﻿namespace Vibe.Domain.SupportRequests
{
    public class SupportRequest
    {
        public Guid Id { get; }
        public String Title { get; }
        public String Description { get; }
        public Guid ClientId { get; }
        public Guid? EmployeeId { get; }
        public DateTime OpenedAt { get; set; }
        public Boolean IsClosed { get; }

        public SupportRequest(Guid id, String title, String description, Guid clientId, Guid? employeeId,
            DateTime openedAt, Boolean isClosed)
        {
            Id = id;
            Title = title;
            Description = description;
            ClientId = clientId;
            EmployeeId = employeeId;
            OpenedAt = openedAt;
            IsClosed = isClosed;
        }
    }
}

File: Vibe.Backoffice/vibe.backoffice.client/src/domain/techSupports/supportRequest.ts
export class SupportRequest{
    constructor(
        public id: string,
        public title: string,
        public description: string,
        public clientId: string,
        public employeeId: string | null,
        public openedAt: Date,
        public isClosed: boolean
    ) { }
}

export function mapToSupportRequest(data: any): SupportRequest {
    return new SupportRequest(data.id, data.title, data.description, data.clientId, data.employeeId, new Date(data.openedAt), 
        data.isClosed)    
}

export function mapToSupportRequests(data: any[]): SupportRequest[] {
    return data.map(d => mapToSupportRequest(d))
}